Catch font smoothing on all body elements

acconrad opened the pull request:
`-webkit-font-smoothing` will not catch everything in `body`, this can be verified by creating a `<button>` on a screen next to a `<a class="button">` and noting the difference in font smoothing. This PR will fix font smoothing and keep it consistent across all HTML elements.

tysongach commented:
Is it only the `button` that doesn’t inherit this, or is this a more widespread issue? I only ask because we want to try to reduce wildcard selectors whenever possible…

joshuaogle commented:
This is kind of a tricky one, and is either a bug in Chrome or expected behavior, depending on who you ask. There are a few form elements that don't inherit things like font-smoothing, font-size, and a few other styles: `button`, `input`, `textarea` and `select`.

Text inputs will inherit if you inherit it like this:
```
#{all-text-inputs},
textarea {
  -webkit-font-smoothing: inherit;
  font-family: inherit;
}
```

You can also get buttons, input[type="submit"] and selects to inherit like this, but you have to add in `-webkit-appearance: none;`, which I often do in my own projects but I wouldn't want to add to Bitters.

So I wouldn't call this a bug in Bitters, because this gets into the sticky business of overriding browser behavior, but hopefully that helps you solve the problem in your project.

acconrad commented:
My only concern is that while it may not be a Bitter-specific bug, it does create inconsistencies in font-rendering behavior, which the end user may not be aware of. They shouldn't expect that if they create a link button and a button button next to each other, that they should look different. It should *just work*. To that end, either this PR should be merged or the suggestion by @joshuaogle should be implemented.

whmii commented:
I agree with @tysongach that we reeeeeealy want to avoid wildcard selectors and font-smoothing just isn;t important enough to make an exception to that rule. I would be down for this if it was the following…

```
#{all-text-inputs},
textarea {
  -webkit-font-smoothing: inherit;
  font-family: inherit;
}
```

tysongach commented on app/assets/stylesheets/_typography.scss:
```diff
@@ -1,11 +1,17 @@
+  -webkit-font-smoothing: antialiased;
+  
+  #{$all-text-inputs},
+  textarea {
```
I think these styles should go inside the same ruleset found in [`_forms.scss`](https://github.com/thoughtbot/bitters/blob/master/app/assets/stylesheets/_forms.scss).

tysongach replied:
@acconrad Any update here?

tysongach commented on app/assets/stylesheets/_typography.scss:
```diff
@@ -1,11 +1,17 @@
   font-family: $base-font-family;
   font-size: $base-font-size;
   line-height: $base-line-height;
+  -webkit-font-smoothing: antialiased;
```
:abc: This should be sorted just below `font-size`.

tysongach replied:
@acconrad Any update here?

tysongach commented:
@acconrad Still a few open comments and questions here (see collapsed comments above). Also, can you please rebase against `master` to pull in the latest changes?

acconrad commented:
@tysongach what open comments are there? I moved the font smoothing into the forms file and refactored the typography file to be in the correct attribute order.

tysongach commented:
@acconrad Couple of questions:

1. Your original comment mentions the `button` element, but all changes in this PR are on text inputs (via Bourbon’s [`#{$all-text-inputs}`](http://bourbon.io/docs/#text-inputs) helper). Are you certain these code changes are what you’re expecting?
2. Can you [update your fork](https://robots.thoughtbot.com/keeping-a-github-fork-updated) and rebase against `master`? This PR is out of date with the latest changes to Bitters.
3. Can you post a screenshot of the font smoothing acting differently across certain elements? I’m not sure I’m finding the same thing you are. Here’s some dummy code (styled with a fresh Bitters instance) and a screenshot from my retina MacBook Pro (Chrome on the left, Safari on the right):

```html
<button>Submit</button>
<input type="submit" value="Submit">
<a class="button">Submit</a>
```

![buttons](https://cloud.githubusercontent.com/assets/903327/7033769/b1a485ca-dd4a-11e4-8a4c-ecdb633e454e.jpg)

I don’t see any difference between the three element types, nor across browsers.

acconrad commented:
It appears as if the current webkit browsers no longer have this issue - it must have been only relevant back in December. Closing this unless I find it again
